### PR TITLE
Matt/route detail

### PR DESCRIPTION
**Pull request by mattbarker016**
Improved map drawing
iPhone X improvements
Loading indicator / walking directions
Code quality

### CODE REVIEW

File: TCAT/Controllers/RouteDetailViewController.swift [hunk 1]
```diff
@@ -211,26 +257,102 @@ class RouteDetailViewController: UIViewController, GMSMapViewDelegate, CLLocatio
...
         view = mapView
+        if isLoading { view.alpha = 0 }
+        
     }
+    
+    // MARK: Loading Screen Functions
+    
+    func createLoadingScreen() {
+        
+        loadingView = UIView(frame: navigationController!.view.frame)
```
Comment (AAAstorga) [inline TCAT/Controllers/RouteDetailViewController.swift]: dat force unwrap doe

File: TCAT/Controllers/RouteDetailViewController.swift [hunk 2]
```diff
@@ -322,43 +448,42 @@ class RouteDetailViewController: UIViewController, GMSMapViewDelegate, CLLocatio
...
+        
+        for path in paths {
+            
+            path.traveledPolyline.map = mapView
+            path.map = mapView
+            
+            for waypoint in path.waypoints {
                 
-//                if routePath is WalkPath {
+//                 if routePath is WalkPath {
```
Comment (AAAstorga) [inline TCAT/Controllers/RouteDetailViewController.swift:474]: can this be deleted?

**Review by AAAstorga — APPROVED** [review]
Looks like some juicy updates!

File: TCAT/Controllers/RouteDetailViewController.swift (hunk 2 above)
Comment (mattbarker016) [reply TCAT/Controllers/RouteDetailViewController.swift:474]: nah. Future code stuff

**Pull request merged**